lib/middleman-navtree/helpers.rb

```diff
@@ -11,12 +11,21 @@ def tree_to_html(value, depth = Float::INFINITY, key = nil, level = 0)
           # This is a file.
           # Get the Sitemap resource for this file.
           # note: sitemap.extensionless_path converts the path to its 'post-build' extension.
-          this_resource = sitemap.find_resource_by_path(sitemap.extensionless_path(value))
-          # Define string for active states.
-          active = this_resource == current_page ? 'active' : ''
-          title = discover_title(this_resource)
-          link = link_to(title, this_resource)
-          html << "<li class='child #{active}'>#{link}</li>"
+          
+          # Make sure the extension path ends with .html (in case we're parsing someting like .adoc)
+          extensionlessPath = sitemap.extensionless_path(value)
+          unless extensionlessPath.end_with? ".html"
+           extensionlessPath << ".html"
+          end
```

Comment:
I'm trying to think if there are any situations where we wouldn't want to assume that a page ought to turn into HTML. Dotfiles maybe? I'm not sure how those situations are handled currently...